src/sdk/index.js

```diff
@@ -4,23 +4,42 @@ import {merge} from 'lodash'
 import api from './api'
 import {DEFAULT_HEADERS} from './headers'
 
+function configGlobalSpec (globalSpec) {
+
+  let headers
+  if (typeof globalSpec.headers === 'function') {
+    headers = {...globalSpec.headers()}
+  }
+  else {
+    headers = globalSpec.headers
+  }
+
+  return {
+    ...globalSpec,
+    ...{headers}
+  }
+}
+
 export default function sdk (globalSpec = {}) {
 
   let defaultSpec = {
     origin: '',
     headers: DEFAULT_HEADERS
   }
-  let options = merge({}, defaultSpec, globalSpec)
+
+  let options = merge({}, defaultSpec, configGlobalSpec(globalSpec))
+
   let stream = Observable.create()
 
   function apiFactory (instanceSpec = {}) {
 
-  	return api(merge({}, options, instanceSpec, {
```

Comment:
Because `globalSpec` also contains the `x-entity` and `authorization` I don't think it is sufficient to return just `globalSpec.headers()` since we will then lose the `x-entity` and `authorization` fields. It does seem like there are some redundancies though and those should be simplified.